addons/web/controllers/main.py

```diff
@@ -426,14 +426,30 @@ def xml2json_from_elementtree(el, preserve_whitespaces=False):
 
 def binary_content(xmlid=None, model='ir.attachment', id=None, field='datas', unique=False,
                    filename=None, filename_field='datas_fname', download=False, mimetype=None,
-                   default_mimetype='application/octet-stream', related_id=None, access_mode=None, access_token=None,
+                   default_mimetype='application/octet-stream', access_token=None,
                    env=None):
     return request.registry['ir.http'].binary_content(
         xmlid=xmlid, model=model, id=id, field=field, unique=unique, filename=filename,
         filename_field=filename_field, download=download, mimetype=mimetype,
-        default_mimetype=default_mimetype, related_id=related_id, access_mode=access_mode, access_token=access_token,
+        default_mimetype=default_mimetype, access_token=access_token,
         env=env)
 
+def limited_image_resize(content, width=None, height=None, crop=False, upper_limit=False, avoid_if_small=False):
+    if crop and (width or height):
```

Comment:
Just a little detail here but not a big deal, if content is none and we give it to crop image, we will get False as a result: 
still falsy, but to avoid to change this value, we could check content truthiness

`if content and crop and (width or height):`

More than that:
```
def limited_image_resize(content, width=None, height=None, crop=False, upper_limit=False, avoid_if_small=False):
    if content and (width or height):
        if crop:
            return crop_image(content, type='center', size=(width, height), ratio=(1, 1))
        else content and (width or height):
            if not upper_limit:
                # resize maximum 500*500
                width = min(width, 500)
                height = min(height, 500)
            return image_resize_image(
                base64_source=content, size=(width or None, height or None), encoding='base64', upper_limit=upper_limit,
                avoid_if_small=avoid_if_small)
    return content
```

As Anthony said, we may remove avoid_if_small but it looks like it is used